src/EFCore.Relational/Infrastructure/RelationalModelValidator.cs

```diff
@@ -911,5 +913,114 @@ private void ValidateTPHMapping(IEntityType rootEntityType, bool forTables)
 
         private static string Format(string tableName, string schema)
             => schema == null ? tableName : schema + "." + tableName;
+
+        /// <summary>
+        ///     Validates that the properties of any one index are
+        ///     all mapped to columns on at least one common table.
+        /// </summary>
+        /// <param name="model"> The model to validate. </param>
+        /// <param name="logger"> The logger to use. </param>
+        protected virtual void ValidateIndexProperties(
+            [NotNull] IModel model, [NotNull] IDiagnosticsLogger<DbLoggerCategory.Model.Validation> logger)
+        {
+            Check.NotNull(model, nameof(model));
+
+            foreach (var entityType in model.GetEntityTypes())
+            {
+                foreach (var index in entityType.GetIndexes()
```

Comment:
```suggestion
                foreach (var index in entityType.GetDeclaredIndexes()
```

No need to check inherited indexes again

Reply:
This works - must be the comment above that causes the test Detects_index_properties_mapped_to_different_tables_in_TPT_hierarchy to fail.